### PR TITLE
txn: change pessimistic lock keys to primary key first, and then all secon…

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve? 

Change pessimistic lock keys working mode from locking all keys concurrently to primary key first and then secondary keys concurrently, to solve the resolving problems in some scenarios

### Check List 

Tests 

 - Unit test

Code changes

Side effects

 - Possible performance regression

Related changes

Release note

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

**Comment by cfzjywxk** [comment]
@jackysp @tiancaiamao @lysu @youjiali1995  PTAL

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -391,7 +391,8 @@ func (c *twoPhaseCommitter) doActionOnKeys(bo *Backoffer, action twoPhaseCommitA
 	firstIsPrimary := bytes.Equal(keys[0], c.primary())
 	_, actionIsCommit := action.(actionCommit)
 	_, actionIsCleanup := action.(actionCleanup)
-	if firstIsPrimary && (actionIsCommit || actionIsCleanup) {
+	_, actionIsPessimiticLock := action.(actionPessimisticLock)
+	if firstIsPrimary && (actionIsCommit || actionIsCleanup || actionIsPessimiticLock) {
 		// primary should be committed/cleanup first
```
Comment (youjiali1995) [inline store/tikv/2pc.go]: Need to update comments.

**Pull request merged**